### ISSUE TITLE
darktable exhibits anamolies and crashes after restoring data.db and library.db files from snapshots

### ISSUE THREAD

**Issue by Dave22152** [darktable-org/darktable#10559]
**Issue:** When attempting to restore corrupted data.db and library.db files, darktable exhibits anomalies and crashes after restoring from the snapshot files.
The data.db and library.db files were corrupted as a result as a result of a BSOD crash on Windows 10.  I attempted to restore to the most recent snapshot files by renaming data.db-snp-xxxxxxxx and library.db-snp-xxxxxxxx.  While this had succesfully restored DT to the most recently saved catalogue, I experienced the following anamolies:
1. Clicking an image in lightable would not select it
2. Double-clicking an image in lighttable would send the image to darkroom, however the history would only show "0 original image" and all other default modules were missing (see screenshot).  It is possible to invoke new modules and edit the image
3. Images won’t duplicate
4. darktable crashes if I attempt to import additional files (backtrace is attached)

Deleting the files and starting with newly created data and library files restores all functionality without issue.

**To Reproduce**
1.	Delete existing data.db and library.db files
2.	Rename data.db-snp-xxxxxxxx and library.db-snp-xxxxxxxx to data.db and library.db respectively
3.	Restart darktable.  The catalogue saved as of the snapshot date should appear, but with the issues described above.
I believe I have isolated the issue to the renamed library file.
Deleting the renamed files will auto-generate new files and correct the issues, but the catalogue is lost.

**Which commit introduced the error** I found this with DT 3.7 but it happens under 3.6 as well.

**Platform**
darktable version : 3.7.0+1572~g6dc1da83e (but it happens under all my 3.6 versions)
OS : Windows 10
Memory : 8 GB
Graphics card : AMD A8-7410 APU with AMD Radeon R5 Graphics, 2.20 Ghz
Graphics driver : Unk
OpenCL installed : Yes
OpenCL activated : No (Open CL always causes my system to lock up - different issue)
Xorg : Unk
Desktop : HP Pavalion
GTK+ : Unk
gcc : Unk
cflags : Unk
CMAKE_BUILD_TYPE : Unk 

**Additional context**
Given that I found this after a Windows crash, I did run system diagnostics and verified that there were no damaged files or sectors.  Given that I saw this with my 3.6 version, which was not corrupted, I suspect this is a problem within the snapshot files

Note that the history in the attached screen shot only shows "0. Original Image" and no other default modules:

![Screenshot](https://user-images.githubusercontent.com/61152378/144724779-4e864c89-f1c2-4592-b902-05b0b5bb8b15.JPG)

The attached backtrace was generated upon a darktable crash after I attempted to import additional images:

[darktable_bt_2HZGD1.txt](https://github.com/darktable-org/darktable/files/7654704/darktable_bt_2HZGD1.txt)

**Comment by jade-nl** [comment]
I'm not able to reproduce this on my Linux machine (I do not have access to Windows).

If I follow the steps described and start darktable:
- a pop-up shows that tells me that some xml differences have been found, it gives me some options and asks me what I want to do (I selected use newest version),
- I can navigate and select, one or more, images in lighttable view,
- lighttable does show a bunch of skulls, but that is correct. Those images have been removed in the meantime and the snapshot is not aware of that. Running *purge_non_existing_images.sh* takes care of this,
- opening edits in darkroom shows all the edit steps taken,
- making duplicates works,
- adding to the library works.

PS: The snapshot does need to be given the correct permissions, but I do think that @Dave22152 is aware (has been told on pixls about this).
